x-pack/plugins/observability/public/hooks/use_alert_data_view.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { DataView } from '@kbn/data-views-plugin/common';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import { BASE_RAC_ALERTS_API_PATH } from '@kbn/rule-registry-plugin/common';
+import type { ValidFeatureId } from '@kbn/rule-data-utils';
+import useAsync from 'react-use/lib/useAsync';
+import type { AsyncState } from 'react-use/lib/useAsync';
+
+import { ObservabilityAppServices } from '../application/types';
+
+export function useAlertDataView(featureIds: ValidFeatureId[]): AsyncState<DataView> {
+  const { http, data: dataService } = useKibana<ObservabilityAppServices>().services;
+  const features = featureIds.sort().join(',');
+
+  const dataView = useAsync(async () => {
+    const { index_name: indexNames } = await http.get<{ index_name: string[] }>(
+      `${BASE_RAC_ALERTS_API_PATH}/index`,
+      {
+        query: { features },
+      }
+    );
+
+    return await dataService.dataViews.create({ title: indexNames.join(',') });
```

Comment:
nit: no need to `await` 
```suggestion
    return dataService.dataViews.create({ title: indexNames.join(',') });
```

Reply:
I think it is correct the way it is, I think the type is wrong! 
https://github.com/elastic/kibana/blob/53bf927a6fe28545d5b8d7513591f3d8100f8b30/src/plugins/data_views/common/data_views/data_views.ts#L139-L144